Modules/mmapmodule.c

```diff
@@ -89,6 +89,10 @@ my_getpagesize(void)
 #  define MAP_ANONYMOUS MAP_ANON
 #endif
 
+typedef struct {
+    PyTypeObject *mmap_object_type;
```

Comment:
Where the heap type is initialized?
You need to do at mmap_exec

Reply:
Thanks for the review! Updated.

Reply:
By the way, you don't need to define _mmap_state.
See: https://github.com/python/cpython/pull/114674#pullrequestreview-1847536472